tests/unit/nupic/algorithms/nab_detector_test.py

```diff
@@ -0,0 +1,273 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+"""
+Test the NuPIC imports run as expected in 
+nab/detectors/numenta/numenta_detector.py. They are 
+nupic/algorithms/anomaly_likelihood and
+nupic/frameworks/opf/modelfactory.ModelFactory. The intent here is not to test
+functionality but rather that the functions are able to run in NAB.
+
+NAB repo: https://github.com/numenta/NAB
+"""
+
+import copy
+import csv
+import datetime
+import os
+import unittest2 as unittest
+
+from nupic.algorithms import anomaly_likelihood as an
+from nupic.frameworks.opf.modelfactory import ModelFactory
+from nupic.support.unittesthelpers.testcasebase import TestCaseBase
+
+
+def _getDateList(numSamples, startDatetime):
+  """
+  Generate a sequence of sample dates starting at startDatetime and incrementing
+  every 5 minutes.
+  
+  @param numSamples       (int)         number of datetimes to generate
+  @param startDatetime    (datetime)    the start (first) datetime
+  @return dateList        (list)        generated sequence of datetimes
+  """
+  dateList = []
+  td = datetime.timedelta(minutes=5)
+  currentDate = startDatetime + td
+  for _ in xrange(numSamples):
+    dateList.append(currentDate)
+    currentDate = currentDate + td
+  
+  return dateList
+
+
+def _addSampleData(numSamples=20, spikeValue=1.0, spikePeriod=10):
+  """
+  Add sample anomaly data to the existing/new data list. Data is constant 0.0,
+  where anomalies are spikes to 1.0 at an interval set by spikePeriod. The test
+  data is trivial, as explicit testing of functions is done in other unit tests.
+
+  @param numSamples   (int)       number of data entries to produce
+  @param spikeValue   (float)     value of the anomaly spikes
+  @param spikePeriod  (int)       periodicity of anomaly spikes, where one will
+                                  occur for every spikePeriod data entries
+  @return data        (list)      list of generated data entries
+  """
+  # Generate datetimes
+  lastDate = datetime.datetime(2015, 4, 1)
+  dateList = _getDateList(numSamples, lastDate)
+
+  # Generate data with anomaly spikes
+  data = []
+  for idx, date in enumerate(dateList):
+    if (spikePeriod > 0) and ( (idx + 1) % spikePeriod == 0):
+      data.append([date, idx, spikeValue])
+    else:
+      data.append([date, idx, 0.0])
+
+  return data
+
+
+def _writeToCSV(data, headers, fileName):
+  """
+  Write list of data to CSV.
+  
+  @param data       (list)      list of data entries, where each row is a list
+  @param headers    (list)      column headers, where each entry in list is
+                                a string
+  """
+  with open(fileName, "wb") as f:
+    writer = csv.writer(f, delimiter=",", lineterminator="\n")
+    writer.writerows([headers])
```

Comment:
Because `writer.writerow(foo)` iterates over the argument `foo`; each element in `foo` is written to subsequent rows in the csv. So with `foo = [["abc", "def"]]`, "abc" and "def" each get their own cell in the same row of the csv. However, with `foo = ["abc", "def"]`, the first written row would be "a", "b", "c", each in separate cells; the next row would similarly be "d", "e", "f". This is why `writer.writerows(data)` works for line 102 instead of explicitly looping through `data`.